utilities/src/main/java/io/syndesis/qe/utils/SQSUtils.java

```diff
@@ -0,0 +1,147 @@
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.Message;
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName;
+import software.amazon.awssdk.services.sqs.model.QueueDoesNotExistException;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequestEntry;
+
+@Slf4j
+@Component
+@Lazy
+public class SQSUtils {
+    private static final String AWS_ACCOUNT_ID = "928544467084";
+
+    private static String queueUrlPrefix;
+    private static String queueArnPrefix;
+
+    private SqsClient client;
+
+    @PostConstruct
+    public void initClient() {
+        log.info("Initializing SQS client");
+
+        final Account sqs = AccountsDirectory.getInstance().getAccount(Account.Name.AWS)
+            .orElseThrow(() -> new IllegalArgumentException("Unable to find SQS account"));
+        final String region = sqs.getProperty("region").toLowerCase().replaceAll("_", "-");
+
+        client = SqsClient.builder().region(Region.of(region))
+            .credentialsProvider(() -> AwsBasicCredentials.create(sqs.getProperty("accessKey"), sqs.getProperty("secretKey"))).build();
+
+        queueUrlPrefix = String.format("https://sqs.%s.amazonaws.com/%s/", region, AWS_ACCOUNT_ID);
+        queueArnPrefix = String.format("arn:aws:sqs:%s:%s:", region, AWS_ACCOUNT_ID);
+    }
+
+    /**
+     * Gets the URL of the given queue.
+     * @param queue queue name
+     * @return queue url
+     */
+    private String getQueueUrl(String queue) {
+        return queueUrlPrefix + queue;
+    }
+
+    /**
+     * Gets the ARN of the given queue.
+     * @param queue queue name
+     * @return queue ARN
+     */
+    public static String getQueueArn(String queue) {
+        return queueArnPrefix + queue;
+    }
+
+    @PreDestroy
+    public void closeClient() {
+        log.info("Closing SQS client");
+        client.close();
+    }
+
+    /**
+     * Sends one or more messages as batch message. The messages have generated IDs and the body as specified.
+     *
+     * @param messageContents bodies of the messages to send
+     */
+    public void sendMessages(String queueName, List<String> messageContents) {
+        log.debug("Sending batch message to queue " + queueName + " with messages: " + messageContents.toString());
+        SendMessageBatchRequest.Builder sendMessageBatchRequestBuilder = SendMessageBatchRequest.builder().queueUrl(getQueueUrl(queueName));
+        List<SendMessageBatchRequestEntry> entries = new ArrayList<>();
+        for (int i = 1; i < messageContents.size() + 1; i++) {
```

Comment:
I'd probably prefer a standard loop starting at 0 and changing the mod check to `i % 10 == 9` or even `(i + 1) % 10 == 0`. I think there's less of a chance of introducing bugs when somebody needs to touch this code in the future.

(also easier at a glance to see what's happening)

Maybe even use something like `Iterables.partition` to split it to chunks beforehand and avoid this loop altogether

Reply:
cool, I didn't know the `Iterables.partition`, I'll edit